packages/teleport/src/Teleport.tsx

```diff
@@ -44,9 +44,16 @@ const Teleport: React.FC<Props> = ({ ctx, history, children }) => {
           <Switch>
             <Route
               title="Login Failed"
-              path={cfg.routes.loginFailed}
-              component={LoginFailed}
+              path={cfg.routes.loginError}
+              render={({ location }) => (
+                <LoginFailed
+                  isCallbackErr={
+                    location.pathname === cfg.routes.loginErrorCallback
+                  }
+                />
+              )}
```

Comment:
```tsx
export default function Container() {
  return (
    <Switch>
      <Route path={cfg.routes.loginErrorCallback}>
        <LoginFailed message="unable to process callback"/>
      </Route>
      <Route component={LoginFailed} />
    </Switch>
  );
}

export function LoginFailed() {
  const message = getUrlParameter('details', window.location.search);
  return (
    <>
      <LogoHero />
      <CardFailed loginUrl={cfg.routes.login} message={message} />
    </>
  );
}

```

then

```tsx
  import LoginFailed from 'xxx';

 <Route path={cfg.routes.loginError} component={LoginFailed} />
 <Route path={cfg.routes.loginErrorLegacy} component={LoginFailed} />
 ```
 
 LoginFailed component will handle its own internal routes without exposing its logic to Teleport.tsx

Reply:
revised